x-pack/plugins/files/server/blob_storage_service/adapters/es/es.ts

```diff
@@ -48,15 +54,64 @@ export class ElasticsearchBlobStorage implements BlobStorage {
     }
   }
 
-  async upload(fileName: string, content: ReadStream): Promise<{ uri: string }> {
-    throw new Error('Not implemented');
+  async upload(src: Readable): Promise<{ id: string; size: number }> {
+    try {
+      await this.createIndexIfNotExists();
```

Comment:
Checking if index exists before every file upload is probably not something we want to do. How else can we do this? Create the index when plugin starts? And then here instead await for that task:

```suggestion
      await this.createIndexOnStartupPromise;
```

Reply:
That is how I did it initially, but then I ran into concurrency issues with starting multiple Kibana's at once. Even catching errors for "already exists" still caused issues in CI - although I suppose we can dig into this.

FWIW, we are only checking if the index exists once but lazily in the current implementation.